### ISSUE TITLE
Crash during testing when state.invalid is set

### ISSUE THREAD

**Issue by david-pfx** [Auroriax/PuzzleScriptPlus#87]
In a test such as "each pattern to match on the left must have a corresponding pattern on the right of equal length" 
* the state.invalid value is set true
* Later loadFile() returns null
* At compiler.js 3202 state is set to null.
* At compiler.js state is compared to undefined (not null)
* Code at 3251 is executed and (of course) it fails.
The proposed fix is to replace line 3250 by:
`        if (IDE && state) {`
Any test that sets state.invalid will trigger this error.

**Comment by david-pfx** [comment]
BTW the code testing for PS+ errors does not belong here. But that's another story.

**Comment by Auroriax** [comment]
I'm sorry but I'm not following. What are you attempting to do that causes these issues? Could you include a repro case perhaps?

**Comment by david-pfx** [comment]
The above is the repro case. The code is visibly wrong. The fix is trivial. I've included it in my merge.

**Comment by Auroriax** [comment]
I'm sorry but I still don't understand what the issue is.
- What is "test" exactly in the context of "In a test such as "each pattern to match on the left must have a corresponding pattern on the right of equal length"?
- Do you have a specific project (or test, guessing from the context) that is causing issues?

**Comment by david-pfx** [comment]
This is the title of one of the tests in the PS suite. Hence the quote marks. It has no meaning (to me) otherwise.

You can see it by simply running the PS tests, or the PR I submitted (which includes it).

**Comment by david-pfx** [comment]
a) Just read the code. It's obviously wrong, even if the tests didn't show it.
b) Are  you running the tests in the debugger and watching the console? In both PS and PS+ many tests fail with a type error, but the test report as passed. This is a separate issue.

**Comment by Auroriax** [comment]
Right, I can reproduce, and I can also reproduce this in the latest version of vanilla PS too (a6db028 - fixes #951), though I'm not sure if it's failing on the same tests. Still, this likely means it's a vanilla PS bug, please consider also reporting it there.

I'm trying to find line 3250 in compile.js, but the file only goes to line 31XX or so. Can you double check if the line number is correct? At this point you can also submit a PR, that might be clearer for me.

**Comment by david-pfx** [comment]
I have already submitted to PS: https://github.com/increpare/PuzzleScript/issues/961. 
The fix and that line no are already in my response to #23.
